src/add-text-track-data.js

```diff
@@ -99,11 +99,9 @@ const addTextTrackData = function(sourceHandler, captionArray, metadataArray) {
       for (let i = 0; i <= sourceHandler.metadataTrack_.cues.length - 1; i++) {
         if (i === sourceHandler.metadataTrack_.cues.length - 1) {
           sourceHandler.metadataTrack_.cues[i].endTime = videoDuration;
-        } else if (sourceHandler.metadataTrack_.cues[i].endTime !==
-          sourceHandler.metadataTrack_.cues[i + 1].startTime &&
-          i < sourceHandler.metadataTrack_.cues.length - 1) {
+        } else {
           sourceHandler.metadataTrack_.cues[i].endTime =
-            sourceHandler.metadataTrack_.cues[i + 1].startTime;
+          sourceHandler.metadataTrack_.cues[i + 1].startTime;
```

Comment:
Looking at this function in it's entirety I think it might help to remove the odd condition from the loop and change the whole thing into:

for (let i = 0; i <= sourceHandler.metadataTrack_.cues.length - 1; i++) {
@imbcmdth     Video.js member
imbcmdth added a line comment 2 minutes ago
Usually, this is better written as for (let i = 0; i < sourceHandler.metadataTrack_.cues.length; i++) {. Most developers are used to this and will automatically read it as "for each element in the array".

EDIT:

Actually looking at the whole for loop I think it makes more sense to do what I talk about in the comment below...

```javascript
let cues = sourceHandler.metadataTrack_.cues;

for (let i = 0; i <cues.length - 1; i++) {
      cues[i].endTime = cues[i + 1].startTime;
}

cues[cues.length - 1].endTime = videoDuration;
```